metals/src/main/scala/scala/meta/internal/metals/ServerCommands.scala

```diff
@@ -112,39 +112,65 @@ object ServerCommands {
        |""".stripMargin
   )
 
-  val DiscoverTestSuites = new Command(
-    "discover-test-suites",
-    "Discover test suites",
-    """|Discovers test suites in project
-       |
-       |The response will be an Array of `SuiteDiscovery`.
-       |
-       |```ts
-       |interface SuiteDiscovery {
-       |  targetName: string;
-       |  targetUri: string;
-       |  discovered: TestDiscoveryResult[];
-       |}
-       |```
-       |```ts
-       |type TestDiscoveryResult = SuiteDiscovery | PackageDiscovery;
-       |```
-       |```ts
-       |interface SuiteDiscovery {
-       |  kind: "suite";
-       |  className: string;
-       |  fullyQualifiedName: string;
-       |  location: Location;
-       |}
-       |```
-       |```ts
-       |interface PackageDiscovery {
-       |  kind: "package";
-       |  prefix: string;
-       |  children: TestDiscoveryResult[];
-       |}
-       |```
-       |""".stripMargin
+  /** If uri is null discover all test suites, otherwise discover testcases in file */
+  final case class DiscoverTestParams(
+      @Nullable uri: String = null
+  )
+  val DiscoverTestSuites = new ParametrizedCommand[DiscoverTestParams](
+    "discover-tests",
+    "Discover tests",
+    """
+      |Discovers all test suites in project
+      |
+      |The response will be an Array of `BuildTargetUpdate`s.
```

Comment:
This seems to be duplicated from the Client command, will we support communication both ways? For sure it's best to have it in one place.

It's a bit weird to havve this kind of mirror protocol, can't we make it one way only?

Reply:
> will we support communication both ways?

Yes, it's needed for single test cases - https://github.com/scalameta/metals/pull/3554#issuecomment-1022034873

Reply:
I moved them to `ClientCommands` since it's more important than `ServerCommands`